File: xssfinder-core/src/main/java/org/xssfinder/runner/DriverWrapper.java
package org.xssfinder.runner;

import org.xssfinder.xss.XssGenerator;

import java.util.Map;
import java.util.Set;

/**
 * Interface that adaptors wrapping different website renderers (like WebDriver) must implement
 */
public interface DriverWrapper {
    /**
     * @return A PageInstantiator that can create pages driven by the wrapped driver
     */
    PageInstantiator getPageInstantiator();

    /**
     * Navigate the driver to given URL
     *
     * @param url The URL to visit
     */
    void visit(String url);

    /**
     * Put XSS attacks from the given XssGenerator into all available inputs
     *
     * @param xssGenerator XssGenerator to produce XSS attacks
     * @return A map of input identifiers -> attack identifiers
     */
    Map<String, String> putXssAttackStringsInInputs(XssGenerator xssGenerator);

    /**
     * @return The set of currently XSS attack identifiers observable on the current page
     */
    Set<String> getCurrentXssIds();

    /**
     * @return The number of forms observable on the current page
     */
    int getFormCount();
}

File: xssfinder-php-executor/src/XssFinder/Runner/DriverWrapper.php
<?php

namespace XssFinder\Runner;

interface DriverWrapper
{
    /**
     * @return PageInstantiator A PageInstantiator that can create pages driven by the wrapped driver
     */
    function getPageInstantiator();

    /**
     * Navigate the driver to given URL
     *
     * @param $url string The URL to visit
     */
    function visit($url);

    /**
     * Put XSS attacks from the given XssGenerator into all available inputs
     *
     * @param $xssGenerator XssGenerator XssGenerator to produce XSS attacks
     * @return array A map of input identifiers -> attack identifiers
     */
    function putXssAttackStringsInInputs($xssGenerator);

    /**
     * @return array The set of currently XSS attack identifiers observable on the current page
     */
    function getCurrentXssIds();

    /**
     * @return int The number of forms observable on the current page
     */
    function getFormCount();

    /**
     * Close the existing session and open a new one, ready for the next test
     */
    function renewSession();
}